### PR TITLE
feat: support to open actions dropdown with keyboard shortcut

### PR DESCRIPTION
**Pull request by micaelagoffe**
Add support to open actions dropdown with keyboard shortcut (shift + a).

[AB#14428](https://dev.azure.com/neovici/Cosmoz3/_workitems/edit/14428/)

### CODE REVIEW

File: src/cosmoz-bottom-bar-focused-context.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	const [bars, setBars] = useState([]),
+		remove = (el) =>
+			setBars((prevBars) => prevBars.filter((bar) => bar !== el)),
+		add = (el) => {
+			setBars((prevBars) => [el, ...prevBars]);
+			return () => remove(el);
+		},
+		focusHandler = useMemo(() => {
+			return { bars, add, remove };
+		}, [bars]);
```
Comment (cristinecula) [inline src/cosmoz-bottom-bar-focused-context.js:23]:
```suggestion
		focusHandler = useMemo(() => ({ bars, add, remove }), [bars]);
```

File: src/cosmoz-bottom-bar-focused-context.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { createContext, useContext, useState, useMemo } from '@pionjs/pion';
+import { noop } from '@neovici/cosmoz-utils/function';
+
+export const bottomBarFocused = createContext({
```
Comment (cristinecula) [inline src/cosmoz-bottom-bar-focused-context.js:4]: We should drop the `bottomBar` prefix from every name in this file. We are *in* the bottom-bar package, there's no need to prefix things.

```suggestion
export const Focus = createContext({
```

File: src/cosmoz-bottom-bar-focused-context.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	bars: [],
+	add: noop,
+	remove: noop,
+});
+customElements.define('bottom-bar-focused-provider', bottomBarFocused.Provider);
```
Comment (cristinecula) [inline src/cosmoz-bottom-bar-focused-context.js:9]: We do however need to prefix the element name, as that is registered globally.

```suggestion
customElements.define('cosmoz-bottom-bar-focus-provider', Focus.Provider);
```

File: src/cosmoz-bottom-bar-focused-context.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+export const useBottomBarFocusedCtx = () => useContext(bottomBarFocused) ?? {};
+
+export const useBottomBarFocusedCtxHandler = () => {
```
Comment (cristinecula) [inline src/cosmoz-bottom-bar-focused-context.js:13]:
```suggestion
export const useFocusHandler = () => {
```

File: src/cosmoz-bottom-bar-focused-context.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import { noop } from '@neovici/cosmoz-utils/function';
+
+export const bottomBarFocused = createContext({
+	bars: [],
+	add: noop,
+	remove: noop,
+});
+customElements.define('bottom-bar-focused-provider', bottomBarFocused.Provider);
+
+export const useBottomBarFocusedCtx = () => useContext(bottomBarFocused) ?? {};
```
Comment (cristinecula) [inline src/cosmoz-bottom-bar-focused-context.js:11]: We don't need this.

**Review by cristinecula — CHANGES_REQUESTED** [review]

File: src/cosmoz-bottom-bar-focused-context.js (hunk 2 above)
Comment (micaelagoffe) [reply src/cosmoz-bottom-bar-focused-context.js:4]: I will import `useBottomBarFocusedCtxHandler` in cosmoz-frontend, so I'm keeping the bottomBar prefix for that function?

File: src/cosmoz-bottom-bar-focused-context.js (hunk 4 above)
Comment (micaelagoffe) [reply src/cosmoz-bottom-bar-focused-context.js:13]: Remove bottomBar prefix here even if I'll import it in cosmoz-frontend?

File: src/cosmoz-bottom-bar-focused-context.js (hunk 5 above)
Comment (micaelagoffe) [reply src/cosmoz-bottom-bar-focused-context.js:11]: What is it we don't need? It doesn't work without `export const useBottomBarFocusedCtx = () => useContext(bottomBarFocused) ?? {};`?

**Comment by cristinecula** [comment]
Generalized the behavior in https://github.com/Neovici/cosmoz-utils/pull/191
We'll need another PR to use that mechanism when it is available.

**Pull request closed without merging**